components/ai_chat/core/browser/ai_chat_service.cc

```diff
@@ -544,10 +548,44 @@ void AIChatService::MaybeAssociateContentWithConversation(
       associated_content_id, conversation->get_conversation_uuid());
 }
 
+void AIChatService::GetNoticesState(GetNoticesStateCallback callback) {
+  bool has_user_dismissed_storage_notice =
+      profile_prefs_->GetBoolean(prefs::kUserDismissedStorageNotice);
+  base::Time last_accepted_disclaimer =
+      profile_prefs_->GetTime(ai_chat::prefs::kLastAcceptedDisclaimer);
+
+  bool is_user_opted_in = !last_accepted_disclaimer.is_null();
+
+  // Premium prompt is only shown conditionally (e.g. the user hasn't dismissed
+  // it and it's been some time since the user started using the feature).
+  bool can_show_premium_prompt =
+      !profile_prefs_->GetBoolean(prefs::kUserDismissedPremiumPrompt) &&
+      !last_accepted_disclaimer.is_null() &&
+      last_accepted_disclaimer < base::Time::Now() - base::Days(1);
+
+  bool is_storage_enabled = profile_prefs_->GetBoolean(prefs::kStorageEnabled);
+
+  std::move(callback).Run(is_user_opted_in, is_storage_enabled,
+                          has_user_dismissed_storage_notice,
+                          can_show_premium_prompt);
+}
+
 void AIChatService::MarkAgreementAccepted() {
   SetUserOptedIn(profile_prefs_, true);
 }
 
+void AIChatService::EnableStoragePref() {
+  profile_prefs_->SetBoolean(prefs::kStorageEnabled, true);
+}
+
+void AIChatService::DismissStorageNotice() {
```

Comment:
We reach this if the user presses X on the top-right of the `Conversation history` popup in Leo, right? We also need an explicit "Don't do this" or "No thanks" button on that popup, otherwise this seems like a consent dark pattern. The language on that popup right now makes me think that the X will actually just dismiss the popup without actually opting me out of this feature change.

Reply:
@ShivanKaul The storage notice is dismissed either by clicking the X or after it's been viewed once for a few seconds.
As far as I understand, the opinion was that we did not need to present the opt-out preference inline as it's not going to affect any previous actions the user does, only the future actions. And that we don't need explicit opt-in permission from the user to store submitted conversations on their own computer (encrypted) for the AI chat feature. This seems more of a notice about the new feature. @mattmcalister